CRM/Utils/System/Backdrop.php

```diff
@@ -419,6 +424,44 @@ public function isPasswordUserGenerated() {
     return TRUE;
   }
 
+  /**
+   * @inheritdoc
+   */
+  public function verifyPassword($params, &$errors) {
+    $errors = form_get_errors();
```

Comment:
While looking at the other PR I noticed that checkUserNameEmailExists also does this, which effectively clears any errors previously set in the formRule function, because they haven't yet been passed back to the form system. I can give more detail if needed.

Reply:
Yes, I copied it from that method. Maybe both are problematic. Maybe instead it can be changed in both methods to add to the `$errors` array rather than replace it. It would be good to make them independent so as not to assume the one is run before the other. Something like:

```
    if ($backdrop_errors = form_get_errors()) {
      // unset Backdrop messages to avoid twice display of errors
      unset($_SESSION['messages']);
      $errors = array_merge($errors, $backdrop_errors);
    }
```

Reply:
Yep something like that sounds good.